builder/build_validate.go

```diff
@@ -255,14 +257,34 @@ func (b *Builder) mcaPkgInfo(mInfo map[string]*swupd.Manifest, version, download
 		return nil, err
 	}
 
+	if repoURLOverrides != nil {
+		// Create tmp DNF conf with overridden baseurl values for specified repos
+		tmpConf, err := ioutil.TempFile(os.TempDir(), "mixerTmpConf")
+		if err != nil {
+			return nil, err
+		}
+		defer func() {
+			_ = os.Remove(tmpConf.Name())
```

Comment:
You should also close the file descriptor before removing the file.

Reply:
Is it not safe to rely on the garbage collector for this?

Reply:
the GC runs concurrent to the main program, so we cannot ensure that the file descriptor will be closed before it is reused in some other function.